modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -208,6 +218,56 @@ public void testAddCustomObjectField() throws Exception {
 						).build())));
 		}
 
+		// Object field encrypted missing required properties
+
+		String algorithm = "AES";
+
+		KeyGenerator keyGenerator = KeyGenerator.getInstance(algorithm);
+
+		keyGenerator.init(128);
+
+		Key key = keyGenerator.generateKey();
+
+		_testWithEncryptedObjectFieldProperties(
+			algorithm, true, Base64.encode(key.getEncoded()),
+			() -> AssertUtils.assertFailure(
+				ObjectFieldBusinessTypeException.class,
+				"Business type encrypted can only be used in object " +
+					"definitions with a default storage type",
+				() ->
+					ObjectDefinitionTestUtil.addObjectDefinitionWithStorageType(
+						_objectDefinitionLocalService,
+						ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE,
+						Arrays.asList(
+							new EncryptedObjectFieldBuilder(
+							).labelMap(
+								LocalizedMapUtil.getLocalizedMap(
+									RandomTestUtil.randomString())
+							).name(
+								"a" + RandomTestUtil.randomString()
+							).build()))));
+
+		_testWithEncryptedObjectFieldProperties(
+			algorithm, true, "",
+			() -> AssertUtils.assertFailure(
+				ObjectFieldBusinessTypeException.class,
+				"Encryption key is required for business type encrypted",
+				_addObjectDefinitionWithEncryptedField()));
+
+		_testWithEncryptedObjectFieldProperties(
+			"", true, "",
+			() -> AssertUtils.assertFailure(
+				ObjectFieldBusinessTypeException.class,
+				"Encryption algorithm is required for business type encrypted",
+				_addObjectDefinitionWithEncryptedField()));
+
+		_testWithEncryptedObjectFieldProperties(
+			"", false, "",
+			() -> AssertUtils.assertFailure(
+				ObjectFieldBusinessTypeException.class,
+				"Business type encrypted is disabled",
+				_addObjectDefinitionWithEncryptedField()));
```

Comment:
@pedro-oliveira446 the assertFailure should outside the _testWithEncryptedObjectFieldProperties to follow the pattern, for example:

```java
AssertUtils.assertFailure(
	ObjectFieldBusinessTypeException.class,
	"Encryption key is required for business type encrypted",
	() -> _testWithEncryptedObjectFieldProperties(
		"AES", true, "",
		() -> _addObjectDefinitionWithEncryptedField()));
```